### ISSUE TITLE
BFF With IdentityServer SPA Login

### ISSUE THREAD

**Issue by biqas** [DuendeSoftware/Support#682]
**Which version of Duende BFF are you using?**

6.2.3

**Which version of .NET are you using?**

7.0

**Describe the bug**

If IdentityServer in combination with BFF is configured to not use the build in login screens but instead of using SPA (Angular) login screens to do an oidc login, then it seems to be that the cookies are not set properly. At least the BFF is complaining that no user token is found.
I could also not find anywhere an example with SPA login and BFF?!

If IdentiyServer used with a non-SPA Login and BFF everything works fine.

**To Reproduce**

Use this for IdentiyServer: https://github.com/DuendeSoftware/Samples/tree/main/IdentityServer/v6/UserInteraction/SpaLoginUi/IdentityServerWithSpaLogin
and put a BFF from here: https://github.com/DuendeSoftware/BFF/tree/main/samples/JS6

**Comment by josephdecock** [comment]
Could you please include the specific error message and any relevant logs?

**Comment by biqas** [comment]
Hello @josephdecock, there are no errors accruing, the problem is, that there is no proper cookie issued to the frontend via the BFF. And if calling a protected API via the Frontend over the BFF it is complaining that no "User Token provided".

**Comment by josephdecock** [comment]
Are you getting a login callback at the bff? Does the http response to that request include any http headers that are trying to set an auth cookie?

**Comment by biqas** [comment]
Yes, the BFF gets a callback and is also setting some cookies, and this is the part wich surprises me that the cookies are not recognized by the BFF if client makes a call.
Can it be, that if Razor pages are used some additional processing is happening in the request/response pipeline so that Identity Server can complete some stuff and if SPA login page is used and the redirect is made back to the BFF server that some stuff is uncomplete which is relevant for the BFF to recognize to store the token properly?

**Comment by josephdecock** [comment]
That sounds like a same-site cookie problem. Brock blogged about it [here](https://brockallen.com/2019/01/11/same-site-cookies-asp-net-core-and-external-authentication-providers/) a while back. It sounds like you're getting the same behavior, and I could see how the SPA UI would change the way that redirects are handled, causing this issue to manifest.

**Comment by biqas** [comment]
I have also tried to disable same site, still exists. If I switch to razor pages instead SPA login page, then everything works and I'm not modifying any settings.

**Comment by josephdecock** [comment]
So the http response includes the set cookie header for the auth cookie, but then the auth cookie is not sent on subsequent requests?

If so, that means some rule about cookie processing is being enforced by the browser such that the browser is either discarding the cookie when it receives it, or not sending it on this particular request. When this happens, browser dev tools will almost always be able to give you more information about why that happened, though sometimes it can be a little difficult to find. That's what I recommend you look at next.

**Comment by biqas** [comment]
I guess I will provide you a full code where you can analyse.

**Comment by josephdecock** [comment]
We need you to keep doing analysis in your environment, because the cause of the problem may well be environmental. Fixing this problem is going to require actually running your code, which is something you're best positioned to do. 

The main questions for you are:
1. What set-cookie http headers are included in the response from the bff callback? I need to see, from the network traffic in the dev tools, what are the raw set-cookie http headers in the response.
2. What happens to each of those cookies in the browser? (Can you see each of those cookies in dev tools? Are there any warnings associated with any of those cookies?)
3. When you make subsequent bff requests that fail, which of those cookies is included in that http request and which are not? 
4. Are there any warnings or errors in the browser's dev tools during this whole process?

**Comment by biqas** [comment]
Hello, I have now cloned these projects from scratch, and it is working fine. Seems to be that I have made some where some changes I have to figure out. My guess now is that I was overriding the cookies somehow. Thanks for the help.

Repos used: 
For IdenityServer: https://github.com/DuendeSoftware/Samples/tree/main/IdentityServer/v6/UserInteraction/SpaLoginUi
For BFF:
https://github.com/DuendeSoftware/Samples/tree/main/IdentityServer/v6/BFF/AngularBffSample
Adjust the hosts and clients redirect uris.
